tests/generation/test_utils.py

```diff
@@ -183,25 +191,25 @@ def _get_encoder_outputs(
     def _greedy_generate(
         self,
         model,
-        input_ids,
-        attention_mask,
         max_length,
+        model_kwargs,
         output_scores=False,
         output_logits=False,
         output_attentions=False,
         output_hidden_states=False,
         return_dict_in_generate=False,
     ):
+        input_name = model_kwargs.pop("input_name")
+        inputs = model_kwargs.pop(input_name)
         logits_process_kwargs, _ = self._get_logits_processor_and_warper_kwargs(
-            input_ids.shape[-1],
+            model_kwargs["attention_mask"].shape[-1],
```

Comment:
`input ids` can now be any type of "main input", like pixel values. Attention mask may be more reliable